tests/System.IO.Pipelines.Tests/TestBufferFactory.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO.Pipelines.Testing;
+using System.Linq;
+using System.Text;
+
+namespace System.IO.Pipelines.Tests
+{
+    internal abstract class TestBufferFactory
```

Comment:
Add one that mimics the pipelines allocation pattern.

Reply:
Adding an empty block at the end of a block with data

Reply:
`BytePerSegmentTestBufferFactory` has an empty block in the beginning, end, and between blocks with data.